src/engines/kotor/gui/guibackground.cpp

```diff
@@ -0,0 +1,142 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  The menu background.
+ */
+
+#include <cfloat>
+
+#include "src/common/util.h"
+#include "src/common/ustring.h"
+
+#include "src/aurora/resman.h"
+#include "src/aurora/types.h"
+
+#include "src/graphics/graphics.h"
+#include "src/graphics/resolution.h"
+#include "src/graphics/aurora/textureman.h"
+
+#include "src/engines/kotor/gui/guibackground.h"
+
+namespace Engines {
+
+namespace KotOR {
+
+GuiBackground::GuiBackground(const Common::UString &type) :
+	Graphics::GUIElement(Graphics::GUIElement::kGUIElementBack), _type(type) {
+
+	_distance = FLT_MAX;
+
+	_screenWidth = GfxMan.getScreenWidth();
+	_screenHeight = GfxMan.getScreenHeight();
+
+	update();
+}
+
+GuiBackground::~GuiBackground() {
+	hide();
+}
+
+void GuiBackground::setType(const Common::UString &type) {
+	_type = type;
+}
+
+void GuiBackground::calculateDistance() {
+}
+
+void GuiBackground::render(Graphics::RenderPass pass) {
+	if (pass == Graphics::kRenderPassTransparent || !_render)
+		return;
+
+	TextureMan.set(_texture);
+
+	glDisable(GL_DEPTH_TEST);
+	glDepthMask(GL_FALSE);
+
+	glMatrixMode(GL_PROJECTION);
+	glLoadIdentity();
+	glOrtho(0.0, _screenWidth, 0.0, _screenHeight, -1.0, 1.0);
+
+	glMatrixMode(GL_MODELVIEW);
+	glLoadIdentity();
+
+	glPushMatrix();
+
+	glBegin(GL_QUADS);
```

Comment:
That's then of course another piece of code that needs to be rewritten once the OpenGL rewrite is further along. :P